account_fiscal_position_rule_purchase/models/purchase.py

```diff
@@ -17,16 +17,18 @@ def _fiscal_position_map(self, **kwargs):
             .apply_fiscal_mapping(**kwargs)
         )
 
-    @api.onchange("partner_id", "dest_address_id", "company_id")
-    def onchange_fiscal_position_map(self):
-
-        kwargs = {
+    def _prepare_fiscal_position_map_kwargs(self):
+        self.ensure_one()
+        return {
             "company_id": self.company_id,
             "partner_id": self.partner_id,
             "partner_invoice_id": self.partner_id,
-            "partner_shipping_id": self.partner_id,
+            "partner_shipping_id": self.dest_address_id,
```

Comment:
Does this make sense?
```suggestion
            "partner_shipping_id": self.dest_address_id or self.partner_id,
```